src/api/java/jni/api_solver.h

```diff
@@ -0,0 +1,56 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Daniel Larraz
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2025 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * The cvc5 Java API.
+ */
+
+#ifndef CVC5__API_SOLVER_H
+#define CVC5__API_SOLVER_H
+#include <cvc5/cvc5.h>
+#include <jni.h>
+
+class ApiSolver : public cvc5::Solver
+{
+ public:
+  /**
+   * Construct an ApiSolver.
+   */
+  ApiSolver(cvc5::TermManager& tm);
+
+  /**
+   * Create a new global reference to the object referred to by the obj
+   * argument, and store the object to be disposed of later.
+   */
+  jobject addGlobalReference(JNIEnv* env, jobject object);
```

Comment:
Perhaps we can add here which kinds of global references needed to be stored globally.
Currently, Oracles and plugins.